server/implementation/src/main/java/io/smallrye/graphql/execution/ExecutionService.java

```diff
@@ -74,15 +74,20 @@ public ExecutionResponse execute(JsonObject jsonInput) {
 
         // ExecutionId
         ExecutionId finalExecutionId = ExecutionId.from(executionIdPrefix + executionId.getAndIncrement());
+        LogPayloadOption payloadOption = LogPayloadOption.OFF;
 
         try {
             String query = context.getQuery();
+            Optional<Map<String, Object>> variables = context.getVariables();
 
             if (query == null || query.isEmpty()) {
                 throw new RuntimeException("Query can not be null");
             }
-            if (config.logPayload()) {
+            if (payloadOption.getPayload().equals("withoutvariable")) {
```

Comment:
"withoutvariable" can be replaced with LogPayloadOption.WITHOUTVARIABLE.getPayload()
(same with withvariable)

This make refactoring easier if we later decide to rename for example.